week3/homework/src/utilities.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+const dataDealer = require('./data/data_dealer');
+const read = dataDealer.read;
+const write = dataDealer.write;
+const shortid = require('shortid');
+const TodoItem = require('./todo_obj_constructor');
+
+const validation = function(req) {
+  const reqBodyLength = Object.keys(req.body).length;
+  const reqToDo = req.body.todo || {};
+  let reqDescription = reqToDo.description || {};
+
+  if (typeof reqDescription === 'string') {
+    reqDescription = reqDescription.trim();
+  }
+
+  if (
+    reqBodyLength !== 1 ||
+        reqDescription === {} ||
+        typeof reqDescription !== 'string' ||
+        reqDescription.length === 0
+  ) {
+    return (reqDescription = '');
+  }
+  return reqDescription;
+};
+
+const findToDo = function(CurrentList, reqId) {
+  let toDo = CurrentList.find(todo => todo.id === reqId);
+  if (toDo === undefined) {
+    throw new Error(`The To-Do item with ID:${reqId} is not existed`);
+  }
+  return toDo;
+};
+
+const createToDo = function(req, res) {
+  const toDoDescription = validation(req);
+  if (toDoDescription === '') {
+    res.status(202).json({ Notification: 'The posted To-Do is not valid' });
+    return;
+  }
+  const todoObject = new TodoItem(shortid.generate(), toDoDescription, false);
+  read('./data/todolist.json', 'utf8')
+    .then(result => {
+      const CurrentList = JSON.parse(result);
+      CurrentList.push(todoObject);
+      const updatedList = JSON.stringify(CurrentList, null, 2);
+      write('./data/todolist.json', updatedList);
+      res.status(201).json({ Notification: 'new To-DO is added' });
+    })
+    .catch(err => res.status(500).json({ Error: err.message }));
+};
+
+const showToDos = function(req, res) {
+  read('./data/todolist.json', 'utf8')
+    .then(result => {
+      if (result === '[]') {
+        res.status(202).json({ Notification: 'The ToDo list is empty!' });
+        return;
+      }
+      res.status(200).send(result);
```

Comment:
what about a content type?

Reply:
ops, in this case I am sending it as text without mentioning that. Now I understand why I get that text style when the ToDos list appears in the PostMan